src/loginpage.js

```diff
@@ -0,0 +1,107 @@
+import { Image, Text, Box, Container, Flex, Heading, Link, Select } from "@chakra-ui/react";
+import MarkdownIt from "markdown-it";
+import React, { useState } from "react";
+import hubsConfig from "./hubs.yaml";
+import { LoginPanel } from "./loginpanel";
+import { Message } from "./message";
+import { Interfaces, Welcome } from "./welcome";
+
+const md = new MarkdownIt();
+
+const pageProperties = window.pageProperties;
+
+const Header = ({ logo, tagline, ...props }) => {
+  return <Flex {...props} direction="column" alignItems="center">
+    <Link href="/" marginBottom={4}>
+      <Image src={logo.img} width={logo.width} />
+    </Link>
+    <Text size="xs"
+      dangerouslySetInnerHTML={{ __html: md.render(tagline) }} />
+  </Flex>
+}
+
+
+const getHub = (domain) => {
+  const defaults = hubsConfig.defaults;
+  let hub = hubsConfig.hubs[domain]
+  Object.keys(defaults).forEach(key => {
+    if (hub[key] === undefined) {
+      hub[key] = defaults[key];
+    }
+  })
+
+  return hub;
+}
+
+const HubChooser = ({ setHub }) => {
+  const domain = location.hostname;
+  const queryDomain = new URLSearchParams(location.search).get('domain');
+  if (queryDomain !== null) {
+    if (domain !== 'localhost' && !domain.endsWith('.netlify.app')) {
```

Comment:
Even when the attack surface has diminished, I think this is still problematic...

Maybe you can use the webpack "mode" to only add this piece in dev mode?

https://webpack.js.org/guides/production/#specify-the-mode

And have something like this...

```javascript
if (process.env.NODE_ENV !== 'production') {
  Yuvi injection tool here... ;-)
}
```